spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestRestDropPurgeTable.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.sql;
+
+import static org.apache.iceberg.CatalogProperties.CATALOG_IMPL;
+import static org.apache.iceberg.CatalogProperties.REST_CATALOG_PURGE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.iceberg.Parameter;
+import org.apache.iceberg.ParameterizedTestExtension;
+import org.apache.iceberg.Parameters;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.inmemory.InMemoryCatalog;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.rest.RESTCatalog;
+import org.apache.iceberg.spark.SparkCatalog;
+import org.apache.iceberg.spark.TestBase;
+import org.apache.spark.sql.catalyst.analysis.NamespaceAlreadyExistsException;
+import org.apache.spark.sql.catalyst.analysis.NoSuchNamespaceException;
+import org.apache.spark.sql.catalyst.analysis.TableAlreadyExistsException;
+import org.apache.spark.sql.connector.catalog.Column;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ParameterizedTestExtension.class)
```

Comment:
I think this should probably still be a part of the other test classes, or a new file called TestSparkCatalogREST 

On a higher point I think looking at this we should just add a new method to the SparkCatalog

```java
  /**
   * Reset the delegate Iceberg Catalog to a test Mock Catalog object
   */
  @VisibleForTesting
  void injectCatalog(Catalog iceberg) {
    this.icebergCatalog = icebergCatalog;
  }
```

Which can use to just inject a mock Catalog into the Spark environment so then the code would look something like

Psuedo Code Follows
```
Spark Config Here
RestCatalog mockCatalog = mock(new RestCatalog)
Spark3Util.loadCatalog("name").injectCatalog(mockCatalog)
sql("drop table purge ...")
verify(mockCatalog).called(drop, purge true)
```

@nastra and @amogh-jahagirdar , I know y'all have written a bunch of REST testing, how does that sound to you?

I am wondering whether we should have parameter for the normal test code that uses Hive, Hadoop, and add in a REST that injects like ^ ... but that may be overkill

Reply:
I don't think we need any of the injection logic. We should rather wait for https://github.com/apache/iceberg/pull/11093 to get in, which would enable much easier testing with Spark + REST catalog

Reply:
I see that #11093 has landed, could you help me to proceed here @nastra? I'm not sure how to set the properties of the rest-catalog in `CatalogTestBase` to include 

```java
REST_CATALOG_PURGE,
Boolean.toString(purge)
```

for my test.

Reply:
@twuebi you'll need to extend `CatalogTestBase´ and then override the `parameters()` method to set the catalog to be tested (where you can provide additional properties). An example can be seen in https://github.com/apache/iceberg/pull/11388/files#diff-7f5041107ee18a22c758b79de43bdfcf8582433710adffcfcedd76685a334081R102-R108

Reply:
@nastra How would we be able to check the side-effects of the Catalog then? I'm also not a big fan of us having to do a whole new configuration for the entire test suite just to test one config property

Reply:
@RussellSpitzer & @nastra not sure how to proceed here

Reply:
@nastra ?